packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -104,26 +106,31 @@ export class RetriableDocumentStorageService implements IDocumentStorageService,
         );
     }
 
-    private checkStorageDisposed(callName: string, error: unknown) {
-        if (this._disposed) {
-            this.logger.sendTelemetryEvent({
-                eventName: `${callName}_abortedStorageDisposed`,
-                fetchCallName: callName, // fetchCallName matches logs in runWithRetry.ts
-            }, error);
-            // pre-0.58 error message: storageServiceDisposedCannotRetry
-            throw new GenericError("Storage Service is disposed. Cannot retry", { canRetry: false });
-        }
-        return;
-    }
-
     private async runWithRetry<T>(api: () => Promise<T>, callName: string): Promise<T> {
-        return runWithRetry(
+        const runResult = await runWithRetry2(
             api,
             callName,
             this.logger,
             {
-                onRetry: (_delayInMs: number, error: unknown) => this.checkStorageDisposed(callName, error),
+                cancel: this._disposedController.signal,
             },
         );
+
+        switch (runResult.status) {
+            case "succeeded":
+                return runResult.result;
+            case "failed":
+                throw runResult.error;
+            case "aborted": {
+                this.logger.sendTelemetryEvent({
+                    eventName: `${callName}_abortedStorageDisposed`,
+                    fetchCallName: callName, // fetchCallName matches logs in runWithRetry.ts
+                });
+                // pre-0.58 error message: storageServiceDisposedCannotRetry
+                throw new GenericError("Storage Service is disposed. Cannot retry", { canRetry: false });
```

Comment:
I think you do not need { canRetry: false} here as it's implied. That said, the other place (attach() call) does not have this property, and differences like that RE handling abort signals worry me - I'd rather see it being done uniformly by runWithRetry() code itself